### PR TITLE
[test plan] Macsec docker restart

### PR DESCRIPTION
**Pull request by parmarkj**
### Description of PR
This test is to ensure macsec is restored when macsec docker container is restarted.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
To add a new test case.

#### How did you do it?
Please refer to the readme file.

#### How did you verify/test it?
Verified/passed on virtual T2 testbed with vsonic neighbors.

Screenshot 2024-09-30 at 11 16 08 PM

#### Supported testbed topology if it's a new test case?
T2

### CODE REVIEW

**Comment by judyjoseph** [comment]
@parmarkj Could you check the macsec testplan : https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testplan/MACsec-test-plan.md - Could you add this test at the appropriate section in the same test plan 

In addition, update the section "How did you verify/test it?" in the PR description with the test run/pass status

File: tests/macsec/test_docker_restart.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+pytestmark = [
+    pytest.mark.macsec_required,
+    pytest.mark.topology('t2')
+]
+
+
+def test_restart_macsec_docker(duthost, ctrl_links, policy, cipher_suite, send_sci):
+    logger.info(duthost.shell(cmd="docker ps", module_ignore_errors=True)['stdout'])
+    duthost.restart_service("macsec")
+    duthost.shell(cmd="docker ps", module_ignore_errors=True)['stdout']
```
Comment (BYGX-wcr) [inline tests/macsec/test_docker_restart.py:18]: Shouldn't this be printed in log?

**Comment by BYGX-wcr** [comment]
The code looks fine now. Yet as Judy said, you should post the test run result in the PR description.

> @parmarkj Could you check the macsec testplan : https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testplan/MACsec-test-plan.md - Could you add this test at the appropriate section in the same test plan
> 
> In addition, update the section "How did you verify/test it?" in the PR description with the test run/pass status

**Comment by BYGX-wcr** [comment]
> The code looks fine now. Yet as Judy said, you should post the test run result in the PR description.
> 
> > @parmarkj Could you check the macsec testplan : https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testplan/MACsec-test-plan.md - Could you add this test at the appropriate section in the same test plan
> > In addition, update the section "How did you verify/test it?" in the PR description with the test run/pass status

@parmarkj , what is your plan now? Is it hard for you to post readable test run result?

**Comment by parmarkj** [comment]
@BYGX-wcr  please find the attached screenshot of the log. Thank you.

**Review by judyjoseph — APPROVED** [review]

**Review by BYGX-wcr — APPROVED** [review]

**Comment by judyjoseph** [comment]
@wangxin could you review and merge. thanks

**Comment by BYGX-wcr** [comment]
@judyjoseph , May you merge this PR. All checks have been completed.

**Pull request merged**